### PR TITLE
[FEATURE] TYPO3 10.4

### PR DESCRIPTION
**Pull request by cjanody**
Allow TYPO3 10.4, PHP 7.4+ and remove vendor directory from the repository.

Run `composer run-script extension-release` to build a local .phar file when you want to build a new TER release.

### CODE REVIEW

**Comment by franzholz** [comment]
The Vendor files can only be removed if you also add a clared README.md file. This must contain a description how the Vendor folder can be regenerated by the user.

**Comment by cjanody** [comment]
Hi,
This must be done by whoever uploads the package on the TER, not by the end user. If it's you, you must run `composer run-script extension-release` before generating the archive. This will generate the required phar file. Not sure this should be mentioned in a readme but if you really want one, I would add it. Let me know.
Regards.

**Comment by franzholz** [comment]
Yes, please generate a README file.
So I or the next maintainer will know exactly what to do.
And also an end user might want to get the latest PHP code of the library. Then he also could delete the library files and clone them from github.

**Comment by cjanody** [comment]
Done

File: ext_emconf.php [hunk 1]
```diff
@@ -18,7 +18,7 @@
     'author_company'   => 'jambage.com',
     'constraints'      => [
         'depends'   => [
-            'php'   => '7.2.0-7.3.99',
+            'php'   => '7.2.0-',
```
Comment (franzholz) [inline ext_emconf.php]: You must set an upper limit. This is necessary to tell other users that it has been tested with this version.

File: composer.json [hunk 2]
```diff
@@ -1,75 +1,79 @@
...
+		{
+			"name": "Cyril Janody",
+			"homepage": "https://www.fsg.ulaval.ca"
+		}
+	],
+	"license": [
+		"GPL-2.0-or-later"
+	],
+	"require": {
+		"typo3/cms-core": "^10.4",
```
Comment (franzholz) [inline composer.json]: Also TYPO3 9.5 shall be supported.

File: ext_emconf.php [hunk 3]
```diff
@@ -4,27 +4,27 @@
...
-        ),
-    ),
-);
+    'author'           => 'PhpSpreadsheet developers, Franz Holzinger',
+    'author_email'     => 'franz@ttproducts.de',
+    'author_company'   => 'jambage.com',
+    'constraints'      => [
+        'depends'   => [
+            'php'   => '7.2.0-7.3.99',
+            'typo3' => '10.4.0-',
```
Comment (franzholz) [inline ext_emconf.php]: Also TYPO3 9.5. Set an upper limit.

File: ext_emconf.php [hunk 4]
```diff
@@ -4,27 +4,27 @@
...
-            'base_cache' => '0.0.3-0.0.0'
-        ),
-    ),
-);
+    'author'           => 'PhpSpreadsheet developers, Franz Holzinger',
+    'author_email'     => 'franz@ttproducts.de',
+    'author_company'   => 'jambage.com',
+    'constraints'      => [
+        'depends'   => [
+            'php'   => '7.2.0-7.3.99',
```
Comment (franzholz) [inline ext_emconf.php]: Also PHP 7.4

**Comment by franzholz** [comment]
I have tried to build the packages. 
What is wrong with it?

https://github.com/franzholz/base_excel/issues/3

**Pull request merged**